tmva/tmva/inc/TMVA/RTensor.hxx

```diff
@@ -486,6 +487,22 @@ inline RTensor<Value_t, Container_t> RTensor<Value_t, Container_t>::Reshape(cons
    return x;
 }
 
+/// \brief Resize tensor
+/// \param[in] shape Shape vector
+/// \returns New RTensor
+/// Resize tensor into new shape
+template <typename Value_t, typename Container_t>
+inline RTensor<Value_t, Container_t> RTensor<Value_t, Container_t>::Resize(const Shape_t &shape)
+{
+   // Create new tensor with the specified shape
+   RTensor <Value_t, Container_t> x(shape);
+   
+   // Copying contents from previous tensor
```

Comment:
This is not correct, because you should copy from the old data only the old size and you are using here the new size.
I think it is enough doing (as in `RTensor::ExpandDims` above:
```
RTensor<Value_t, Container_t> x(*this);
x.fShape = shape;
x.fSize = Internal::GetSizeFromShape(shape);
x.fStrides = Internal::ComputeStridesFromShape(shape, fLayout);
```

Reply:
made the change, thanks!